classes/core/PKPRequest.php

```diff
@@ -395,8 +395,13 @@ public function getServerHost(mixed $default = null, bool $includePort = true):
     public function getProtocol(): string
     {
         if (!isset($this->_protocol)) {
-            $this->_protocol = (!isset($_SERVER['HTTPS']) || strtolower($_SERVER['HTTPS']) != 'on') ? 'http' : 'https';
-            Hook::call('Request::getProtocol', [&$this->_protocol]);
+            // If defined, priorizes X-Forwarded-Proto header (#6851)
+            if (!empty($_SERVER['HTTP_X_FORWARDED_PROTO'])) {
+                $this->_protocol = strtolower($_SERVER['HTTP_X_FORWARDED_PROTO']);
+            } else {
+                $this->_protocol = (!isset($_SERVER['HTTPS']) || strtolower($_SERVER['HTTPS']) != 'on') ? 'http' : 'https';
+                Hook::call('Request::getProtocol', [&$this->_protocol]);
```

Comment:
This removes the hook's ability to override the returned value in all cases, which might be important. I'd suggest promoting this outside of this conditional.